distributed/tests/test_scheduler.py

```diff
@@ -4789,6 +4789,22 @@ async def test_resubmit_different_task_same_key_before_previous_is_done(c, s, de
     For a real world example where this can trigger, see
     https://github.com/dask/dask/issues/9888
     """
+
+    class LogEventVerifier(SchedulerPlugin):
+        def start(self, scheduler):
+            self.seen = False
+
+        def log_event(self, topic, msg):
+            if (
+                topic == "all"
+                and isinstance(msg, dict)
+                and msg.get("action", None) == "update_graph"
+                and msg["key-collisions"] > 0
+            ):
+                self.seen = True
+
+    await c.register_plugin(LogEventVerifier(), name="verifier")
```

Comment:
I find the way over a plugin a little verbose and cumbersome. I suggest one of two approaches

```python

def _match(event):
    _, msg = event
    return (isinstance(msg, dict)
                and msg.get("action", None) == "update_graph"
                and msg["key-collisions"] > 0)

# Option 1: use the scheduler get_events method
assert any(_match(ev) for ev in s.get_events(topic))

# Option 2: subscribe_topic
seen = False

def handler(ev):
    if _match(ev):
        nonlocal seen
        seen = True

client.subscribe_topic(topic, handler)
assert seen
```

Reply:
I like the subscription-based model, adopting it.